### PR TITLE
Backendresultedit

### PR DESCRIPTION
**Pull request by CalMacCQ**
Rewritten the docstring for BackendResult as the previous entry was outdated. Perhaps the part about binary strings is not needed.

### CODE REVIEW

File: pytket/docs/passes.rst [hunk 1]
```diff
@@ -10,3 +10,9 @@ pytket.passes.script
 
 .. automodule:: pytket.passes.script
     :members: compilation_pass_from_script, compilation_pass_grammar
+
+pytket.passes.auto_rebase
```
Comment (cqc-alec) [inline pytket/docs/passes.rst:14]: This shouldn't be part of this PR (I think it's already been merged?) Did you reuse the branch? Github behaves better if you create a new branch from `develop` for each PR.

File: pytket/pytket/backends/backendresult.py [hunk 2]
```diff
@@ -66,22 +66,24 @@ class StoredResult(NamedTuple):
...
-    location in basis vector labelling is stored (e.g. statevector index 3
-    corresponds to bitwise encoding 011, and a mapping {Qubit(2): 0} tells us
-    the 0 in the bitwise encoding corresponds to Qubit(2)).
-
-    This allows results to be requested for the specific subset and ordering of
-    units you require. For instance results for [Bit(0), Bit(1)] in that order
-    can be requested using get_counts([Bit(0), Bit(1)]). Requests for state results
-    must be a permutation of all Qubits, not a subset.
-
+    In the case of a real quantum device or a shots based simulator
```
Comment (cqc-alec) [inline pytket/pytket/backends/backendresult.py:69]:
```suggestion
    In the case of a real quantum device or a shots-based simulator
```

File: pytket/pytket/backends/backendresult.py [hunk 3]
```diff
@@ -66,22 +66,24 @@ class StoredResult(NamedTuple):
...
+    a BackendResult will typically be a collection of measurements (shots and counts).
+    Shots are stored internally as 8-bit integers
+    e.g. the '11' basis state is represented as [192] (192=11000000).
```
Comment (cqc-alec) [inline pytket/pytket/backends/backendresult.py:72]: Better not to mention this -- it's irrelevant to users.

File: pytket/pytket/backends/backendresult.py [hunk 4]
```diff
@@ -66,22 +66,24 @@ class StoredResult(NamedTuple):
...
+
+    Results can also be the output of ideal simulations of circuits.
+    These can take the form of statevectors, unitary arrays or density matrices.
+
+    :param q_bits: Sequence of qubits
+    :param c_bits: Sequence of classical bits
+    :param counts: The counts in the result
+    :param shots: The shots in the result
+    :param state: The resulting statevector (i.e from a statevector simulation)
```
Comment (cqc-alec) [inline pytket/pytket/backends/backendresult.py:81]:
```suggestion
    :param state: The resulting statevector (from a statevector simulation)
```

File: pytket/pytket/backends/backendresult.py [hunk 5]
```diff
@@ -66,22 +66,24 @@ class StoredResult(NamedTuple):
...
+    :param unitary: The resulting unitary operator (i.e from a unitary simulation)
```
Comment (cqc-alec) [inline pytket/pytket/backends/backendresult.py:82]:
```suggestion
    :param unitary: The resulting unitary operator (from a unitary simulation)
```

File: pytket/pytket/backends/backendresult.py [hunk 6]
```diff
@@ -66,22 +66,24 @@ class StoredResult(NamedTuple):
...
+    :param density_matrix: The resulting density matrix
+     (i.e output from a density matrix simulator)
```
Comment (cqc-alec) [inline pytket/pytket/backends/backendresult.py:84]:
```suggestion
     (from a density-matrix simulator)
```

File: pytket/pytket/backends/backendresult.py [hunk 7]
```diff
@@ -66,22 +66,24 @@ class StoredResult(NamedTuple):
...
     :param ppcirc: If provided, classical postprocessing to be applied to all measured
-        results (i.e. shots and counts).
```
Comment (cqc-alec) [inline pytket/pytket/backends/backendresult.py:84]: This needs to be indented for sphinx to format it correctly.

**Review by cqc-alec — CHANGES_REQUESTED** [review]

File: pytket/docs/passes.rst (hunk 1 above)
Comment (CalMacCQ) [reply pytket/docs/passes.rst:14]: Hmm this shouldn't be here, I created a fresh branch. Not sure how I managed that. This has already been merged.

**Review by cqc-alec — APPROVED** [review]
